### PR TITLE
Use ruff for linting and formatting

### LINKED ISSUE #104: [MAINT] Use ruff for linting and formatting

### PR DESCRIPTION
**Pull request by pbarbarant**
Fixes #104. Adds ruff to replace flake8, isort and black.

Follows the same config as `nilearn` for consistency.

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
LGTM so far. Do you want to add more changes ?

File: examples/plot_toy_int_experiment.py [hunk 1]
```diff
@@ -143,7 +144,10 @@
 # MDS of tunning matrices
 corr_tunning = compute_pearson_corr(tuning_pred_run_1, tuning_pred_run_2)
 T_first_part_transformed, T_second_part_transformed = matrix_MDS(
-    tuning_pred_run_1, tuning_pred_run_2, n_components=2, dissimilarity=1 - corr_tunning
+    tuning_pred_run_1,
+    tuning_pred_run_2,
+    n_components=2,
+    dissimilarity=1 - corr_tunning,
```
Comment (emdupre) [inline examples/plot_toy_int_experiment.py:150]: Will ruff complain if this is formatted as 
```suggestion
    dissimilarity=(1 - corr_tunning),
```
? I find it more human-readable, but not worth breaking automation !

**Review by emdupre — APPROVED** [review]
This LGTM, thank you @pbarbarant  !

File: examples/plot_toy_int_experiment.py (hunk 1 above)
Comment (pbarbarant) [reply examples/plot_toy_int_experiment.py]: Both `ruff format` and `ruff check` don't seem to complain so I'm adding it.

**Comment by pbarbarant** [comment]
There are still a lot of issues regarding docstrings and documentation though, I suggest adding D-related rules in a future PR.

**Pull request merged**